src/test/java/org/openhab/tools/analysis/checkstyle/test/MarkDownCheckTest.java

```diff
@@ -0,0 +1,248 @@
+        verifyMarkDownFile("testComplicatedCodeBlocks", expectedMessages);
+    }
+
+    @Test
+    public void testMultilineListItemsSeparation() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testListSeparation", expectedMessages);
+    }
+
+    @Test
+    public void testCodeSectionAtEndOfFile() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testCodeSectionAtEndOfFile", expectedMessages);
+    }
+
+    @Test
+    public void testMultiLineListItem() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testMultiLineListItems", expectedMessages);
+    }
+
+    @Test
+    public void testValidMarkDown() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testValidMarkDown", expectedMessages);
+    }
+
+    @Test
+    public void testLinkAsHeader() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testLinkAsHeader", expectedMessages);
+    }
+
+    @Test
+    public void testReadMeAddedInBuildProperties() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(0, ADDED_README_IN_BUILD_PROPERTIES_MSG);
+        String testDirectoryName = "testAddedReadMeInBuildProperties";
+
+        // The message is logged for the build.properties file
+        verifyBuildProperties(expectedMessages, testDirectoryName);
+    }
+
+    @Test
+    public void testAddedDummyDocInBuildProperties() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        String testDirectoryName = "testAddedDummyDocInBuildProperties";
+        verifyBuildProperties(expectedMessages, testDirectoryName);
+    }
+
+    @Test
+    public void testDocFolderAddedInBuildProperties() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(0, ADDED_DOC_FOLDER_IN_BUILD_PROPERTIES_MSG);
+        String testDirectoryName = "testAddedDocFolderInBuildProperties";
+
+        // The message is logged for the build.properties file
+        verifyBuildProperties(expectedMessages, testDirectoryName);
+    }
+
+    @Test
+    public void testAddedReadmeAndDocInBuildProperties() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(0, ADDED_README_IN_BUILD_PROPERTIES_MSG, 0,
+                ADDED_DOC_FOLDER_IN_BUILD_PROPERTIES_MSG);
+        String testDirectoryName = "testAddedReadmeAndDocInBuildProperties";
+
+        // The message is logged for the build.properties file
+        verifyBuildProperties(expectedMessages, testDirectoryName);
+    }
+
+    private void verifyBuildProperties(String[] expectedMessages, String testDirectoryName)
+            throws IOException, Exception {
+        String testDirectoryRelativePath = README_MD_CHECK_TEST_DIRECTORY_NAME + File.separator + testDirectoryName;
```

Comment:
given you don't need the `testDirectoryRelativePath` somewhere else I would not have stored it